### PR TITLE
Minor docstring updates on binning related plot functions

### PR DESCRIPTION
**Pull request by timhoffm**
## PR Summary

As part of #10148: Docstring update for `Axes.hexbin`, `Axes.hist` and `Axes.hist2d`.

### CODE REVIEW

File: lib/matplotlib/axes/_axes.py [hunk 1]
```diff
@@ -4193,7 +4193,7 @@ def hexbin(self, x, y, C=None, gridsize=100, bins=None,
         If *C* is specified, it specifies values at the coordinate
         (x[i],y[i]). These values are accumulated for each hexagonal
         bin and then reduced according to *reduce_C_function*, which
-        defaults to numpy's mean function (np.mean). (If *C* is
+        defaults to numpy's mean function (`numpy.mean`). (If *C* is
```
Comment (anntzer) [inline lib/matplotlib/axes/_axes.py]: rephrase as
```
defaults to `numpy.mean`
```
? (more concise but just as clear)

**Comment by afvincent** [comment]
*I wonder if that belongs to the scope of this PR*, but I think that the description used for the *bins* parameter in `ax.hist` may be outdated. If I am correct, the related keyword argument will be passed to `numpy.histogram`, which means that any string among {"auto", "fd", "doane", "scott", "rice", "sturges", "sqrt"} should be valid provided that the Numpy version is recent enough, right? Currently, it looks like only "auto" is mentioned.

For example, the following (inspired from the numpy [docs](https://docs.scipy.org/doc/numpy-1.14.0/reference/generated/numpy.histogram.html)) runs fine on my laptop with Matplotlib 2.1.0 and numpy 1.13 (both from conda)
```python
import numpy as np
import matplotlib.pyplot as plt

prng = np.random.RandomState(10)  # deterministic random data
a = np.hstack((prng.normal(size=1000),
               prng.normal(loc=5, scale=2, size=1000)))

non_auto_bins_str = ("fd", "doane", "scott", "rice", "sturges", "sqrt")

fig, axs = plt.subplots(nrows=2, ncols=3, sharex=True)
for ax, bins in zip(axs.flat, non_auto_bins_str):
    ax.hist(a, bins=bins)  # arguments are passed to np.histogram
    ax.set_title("'{}' bins".format(bins))

fig.tight_layout()
fig.show()
```
figure_1

**Comment by tacaswell** [comment]
I am hesitant to document the auto-binning on histogram more than 'with a new enough version of numpy strings will work, see `np.histogram`' until our minimum version of numpy is high enough that we are sure it will work.

**Comment by jklymak** [comment]
How about: 'with a new enough version of numpy, strings that specify a binning strategy are also accepted, see `np.histogram`'.

**Comment by anntzer** [comment]
fwiw that's numpy 1.11.

**Comment by timhoffm** [comment]
Thanks for the hint on the binning strategy. I've now written.

> With Numpy 1.11 or newer, you can alternatively provide a string describing a binning strategy, such as ‘auto’, ‘sturges’, ‘fd’, ‘doane’, ‘scott’, ‘rice’, ‘sturges’ or ‘sqrt’, see numpy.histogram.

I think that's clear and useful, and without the danger that it will not work unexpectedly.

File: lib/matplotlib/axes/_axes.py [hunk 2]
```diff
@@ -4193,9 +4193,8 @@ def hexbin(self, x, y, C=None, gridsize=100, bins=None,
         If *C* is specified, it specifies values at the coordinate
         (x[i],y[i]). These values are accumulated for each hexagonal
         bin and then reduced according to *reduce_C_function*, which
-        defaults to numpy's mean function (np.mean). (If *C* is
-        specified, it must also be a 1-D sequence of the same length
-        as *x* and *y*.)
+        defaults to `numpy.mean`. (If *C* is specified, it must also
```
Comment (anntzer) [inline lib/matplotlib/axes/_axes.py]: two lines earlier: space after comma

File: lib/matplotlib/axes/_axes.py [hunk 3]
```diff
@@ -6553,10 +6553,10 @@ def hist2d(self, x, y, bins=10, range=None, normed=False, weights=None,
 
         Parameters
         ----------
-        x, y: array_like, shape (n, )
+        x, y : array_like, shape (n, )
             Input values
 
-        bins: [None | int | [int, int] | array_like | [array, array]]
+        bins : None or int or [int, int] or array_like or [array, array]
```
Comment (anntzer) [inline lib/matplotlib/axes/_axes.py]: (int, int) or (array, array)?  (feels like tuples, nor arrays)

Comment (timhoffm) [reply lib/matplotlib/axes/_axes.py]: This description came from 
[numpy.histogram2d](https://docs.scipy.org/doc/numpy-1.14.0/reference/generated/numpy.histogram2d.html#numpy.histogram2d). I would leave it like this.

File: lib/matplotlib/axes/_axes.py (hunk 2 above)
Comment (timhoffm) [reply lib/matplotlib/axes/_axes.py]: Done.

**Pull request merged**